src/openai/_utils/_utils.py

```diff
@@ -412,3 +416,49 @@ def json_safe(data: object) -> object:
         return data.isoformat()
 
     return data
+
+
+def is_azure_client(client: object) -> TypeGuard[AzureOpenAI]:
+    from ..lib.azure import AzureOpenAI
+
+    return isinstance(client, AzureOpenAI)
+
+
+def is_async_azure_client(client: object) -> TypeGuard[AsyncAzureOpenAI]:
+    from ..lib.azure import AsyncAzureOpenAI
+
+    return isinstance(client, AsyncAzureOpenAI)
+
+
+def configure_azure_realtime(client: AzureOpenAI, model: str, extra_query: Query) -> tuple[Query, dict[str, str]]:
```

Comment:
nit: is the reason the `configure_azure_realtime` and `configure_azure_realtime_async` functions are in `_utils.py` because of circular imports?

I think it'd be cleaner to define these as private methods on the client classes themselves, then the `__aenter__` method changes are just
```py
if is_async_azure_client(self.__client):
  extra_query, auth_headers = await self.__client._configure_realtime(self.__model, extra_query)
```

Reply:
Good call, that's a lot cleaner. Updated!